sdk/identity/azure-identity/src/azure_cli_credential.cpp

```diff
@@ -154,7 +154,10 @@ AccessToken AzureCliCredential::GetToken(
       try
       {
         return TokenCredentialImpl::ParseToken(
-            azCliResult, "accessToken", "expiresIn", "expiresOn");
+            azCliResult,
+            "accessToken",
+            "expiresIn",
+            std::vector<std::string>{"expires_on", "expiresOn"});
```

Comment:
Please add a comment here, that the order matters. We want to check expires_on first, and only if that fails, fallback to the older expiresOn.

It would also be good to mention in a comment that parsing `expiresOn` field is only needed for versions of `Az CLI` older than v.2.54.0. Any new version will emit the new `expires_on` field.

This way, the source will explain **why** we are doing this, to our future selves :)